src/app-layout/utils/use-drawers.ts

```diff
@@ -72,44 +73,56 @@ export function useDrawers(
     __disableRuntimeDrawers: disableRuntimeDrawers,
   }: InternalDrawerProps & { __disableRuntimeDrawers?: boolean },
   toolsProps: ToolsProps,
-  defaults: AppLayoutState
+  defaults: AppLayoutState,
+  {
+    publicDrawers,
```

Comment:
All the logic is here for public vs. beta drawers. It's actually quite simple. I made publicDrawers take precedence, but I don't think that will really be an issue.